allauth/account/adapter.py

```diff
@@ -192,20 +192,23 @@ def save_user(self, request, user, form, commit=True):
         """
         from .utils import user_username, user_email, user_field
 
-        data = form.cleaned_data
-        first_name = data.get('first_name')
-        last_name = data.get('last_name')
-        email = data.get('email')
-        username = data.get('username')
-        user_email(user, email)
-        user_username(user, username)
-        if first_name:
-            user_field(user, 'first_name', first_name)
-        if last_name:
-            user_field(user, 'last_name', last_name)
-        if 'password1' in data:
-            user.set_password(data["password1"])
-        else:
+        try:
+            data = form.cleaned_data
+            first_name = data.get('first_name')
+            last_name = data.get('last_name')
+            email = data.get('email')
+            username = data.get('username')
+            user_email(user, email)
+            user_username(user, username)
+            if first_name:
+                user_field(user, 'first_name', first_name)
+            if last_name:
+                user_field(user, 'last_name', last_name)
+            if 'password1' in data:
+                user.set_password(data["password1"])
+            else:
+                user.set_unusable_password()
+        except AttributeError:
```

Comment:
I am not very fond of using a try/except around a large block of code while catching a very generic exception. I would prefer to avoid the try altogether and keep exceptions for exceptional situations, not for normal code flow. So I would prefer to introduce `if form:` instead.